nemo_aligner/utils/text_generation_utils.py

```diff
@@ -66,12 +68,16 @@ def get_lengths(self) -> torch.Tensor:
             * the token(s) that ended generation, if any (e.g. the `EOS` token or the token(s) corresponding to
               an element of `sampling_params.end_strings`)
         """
-        lengths = None
+        lengths, is_end = None, None
         if parallel_state.is_pipeline_last_stage():  # only the last stage actually has access to lengths
-            lengths = torch.where(self._end_idx >= 0, self._end_idx + 1, self._context_lengths + self._max_length)
+            is_end = self._end_idx >= 0
+            lengths = torch.where(is_end, self._end_idx + 1, self._context_lengths + self._max_length)
+
             lengths = lengths.to(torch.int64).view((-1, 1))
-        lengths = broadcast_2d_tensor_within_pp(lengths, dtype=torch.int64)
-        return lengths.flatten()
+            is_end = is_end.view(-1, 1)
+
+        lengths = broadcast_2d_tensor_within_pp(lengths, dtype=torch.int64).flatten()
+        return lengths
```

Comment:
Seems to me that those changes aren't actually changing anything, is that correct?

Reply:
ah yes that's right. This was from a previous implementation of the is_end logic.

271d5cc189a8635fad84c76c80c6f095c10a9fbc